src/Microsoft.OData.Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -575,7 +576,7 @@ private void MaterializeResolvedEntry(MaterializerEntry entry, bool includeLinks
             // entries. This keeps this code compatible with V1 behavior.
             this.MaterializeDataValues(actualType, entry.Properties, this.MaterializerContext.UndeclaredPropertyBehavior);
 
-            if (entry.NestedResourceInfos != null)
+            if (entry.NestedResourceInfos.Any())

```

Comment:
If you are no longer testing for null, then there is no reason to test for Any() since the loop inside the if will just not run if there is !.Any()

Reply:
@g2mula Good catch... I should change this to `?.Any()`

Reply:
Fixed